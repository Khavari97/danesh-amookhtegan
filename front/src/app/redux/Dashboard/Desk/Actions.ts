import {DESK_LOADING, DESK_USER_PROJECTS, DESK_USER_REQUESTS, IProjectCard, IRequestedCard} from "./ActionTypes";

export const userProjectAction = (projects: IProjectCard[]) => {
    return {
        type: DESK_USER_PROJECTS,
        payload: projects,
        loading: false,
    }
}

export const userRequestAction = (requests: IRequestedCard[]) => {
    return {
        type: DESK_USER_REQUESTS,
        payload: requests,
        loading: false,
    }
}

export const deskLoading = () => {
    return {
        type: DESK_LOADING,
    }
}