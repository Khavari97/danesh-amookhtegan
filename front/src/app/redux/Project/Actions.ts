import {IProjectStatus, SET_PROJECT_SLUG} from "./ActionTypes";

export const setProjectAction = (slug: string, status: IProjectStatus) => {
    return {
        type: SET_PROJECT_SLUG,
        payload: {
            slug,
            status
        }
    }
}