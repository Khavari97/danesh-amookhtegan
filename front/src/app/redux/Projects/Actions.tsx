import {Dispatch} from "redux";
import {apiClient} from "../../api/_api";
import {Get_Projects} from "./ActionTypes";

export const GetProjects = () =>

    (dispatch: Dispatch) => {
        try {
            apiClient.get("dashboard/projects/", {consoleMessage: true})
                .then((response => {
                    dispatch({
                        type: Get_Projects,
                        projects: response.results
                    })

                })).catch((error => {

            }))

        } catch {

        }


    }


