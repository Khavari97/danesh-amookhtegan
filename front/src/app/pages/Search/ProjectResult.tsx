import React, { Component } from 'react';
import Loading from "../../components/customs/Loading/Loading";
import { Grid } from "../../components/material-ui";
import ProjectNotFound from "../../components/customs/ProjectNotFound/ProjectNotFound";
import { apiClient } from "../../api/_api";
import ProjectCard from "../../components/customs/ProjectCard/ProjectCard";
import { IRootState } from "../../redux/RootReducer";
import { connect } from "react-redux";
import { Filter, searchPage, setSearchText } from "../../redux/Search/Actions";
import { ISearchState } from "../../redux/Search/ActionTypes";
import { withRouter } from "react-router-dom";
import SearchProjectFilter from "./SearchProjectFilter";
import BasicPagination from "../../components/customs/Pagination/BasicPagination";
import SearchInput from "../../components/customs/SearchInput/SearchInput";


const mapStateToProps = (store: IRootState) => {
    return store.search
}

const mapDispatchToProps = (dispatch: any) => {
    return {
        FilterProjects: (status: string, category: string, skill: string, text: string, countPages: number) => dispatch(Filter(status, category, skill, text, countPages)),
        searchPage: (page: number) => dispatch(searchPage(page)),
        setSearchText: (text: string) => dispatch(setSearchText(text)),
    }
}
interface IState {
    data: any;
    page: any;
    number: any;
    text: any;
}

interface IProps extends ISearchState {
    location: any,
    FilterProjects: any;
    searchPage: any;
    setSearchText: any
}

class ProjectResult extends Component<IProps, IState> {
    constructor(props: IProps) {
        super(props);
        this.state = {
            data: [],
            page: 1,
            number: 1,
            text: ""

        }
    }

    handleChange = (event: React.ChangeEvent<unknown>, value: number) => {
        this.props.searchPage(value)
    };


    render() {

        return (
            <>
                <SearchInput data_testid="search-input" text={(event: any) => {
                    if (event.target.value.length >= 3 || event.target.value.length === 0) {
                        this.props.setSearchText(event.target.value)
                    }
                }}
                    default={this.props.text}
                />

                <Grid>
                    <SearchProjectFilter />
                </Grid>

                {
                    this.props.loading ? <Loading /> :
                        <Grid className='animate__animated animate__fadeIn' container spacing={2}>
                            {this.props.projects?.length !== 0 ?
                                this.props.projects?.map((project: any, index: number) => {
                                    return <Grid data-testid="project-card" item xs={12} sm={6} md={4}
                                        key={index}><ProjectCard {...project} /></Grid>
                                }) : <ProjectNotFound />}
                        </Grid>
                }
                <BasicPagination siblingCount={0} boundaryCount={2} page={this.props.page} count={this.props.countPages} onChange={this.handleChange} />
            </>
        );
    }

    componentDidMount() {
        this.props.searchPage(1)
    }

    componentDidUpdate(prevProps: IProps) {
        if (this.props.text !== prevProps.text) {
            this.props.FilterProjects(this.props.project_status, this.props.project_category, this.props.project_skill, this.props.text, this.props.page)
        }
        if (this.props.page !== prevProps.page) {
            this.props.FilterProjects(this.props.project_status, this.props.project_category, this.props.project_skill, this.props.text, this.props.page)
        }
    }


}

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(ProjectResult));
