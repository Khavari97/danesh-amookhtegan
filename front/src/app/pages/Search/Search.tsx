import React, { Component } from 'react';
import { Container, Typography } from "@material-ui/core";
import CustomTabs, { TabBarProps } from "../../components/customs/CustomTabs/CustomTabs";
import ProjectResult from "./ProjectResult";
import UserResult from "./UserResult";
import SearchInput from "../../components/customs/SearchInput/SearchInput";


class Search extends Component<any> {

    SearchTabs: TabBarProps[] = [
        { label: "پروژه", component: <ProjectResult /> },
        { label: "افراد", component: <UserResult /> },
    ]

    render() {
        return (
            <Container>
                {this.props.homeVersion ? "" : <Typography data-testid="h1" variant={"h1"} component={"h1"}>جستجو</Typography>}
                <CustomTabs tabs={this.SearchTabs} />
            </Container>
        )
    }
}


export default Search;