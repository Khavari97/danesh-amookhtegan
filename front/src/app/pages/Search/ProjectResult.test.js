import React from 'react';
import Enzyme, { shallow} from "enzyme";
import EnzymeAdapter from "enzyme-adapter-react-16";
import {Provider} from "react-redux";
import store from "../../redux/store";
import ProjectResult  from "./ProjectResult";
import { render, screen, prettyDOM, act, waitFor, wait, fireEvent } from '@testing-library/react'
import sinon from "sinon";
import renderer from "react-test-renderer";
import configureStore from 'redux-mock-store';
import { Filter, searchPage, setSearchText } from "../../redux/Search/Actions";
import thunk from 'redux-thunk'
import * as actions from "../../redux/Search/Actions"
import { BrowserRouter } from 'react-router-dom'

const middlewares = [thunk]
const mockStore = configureStore(middlewares);

Enzyme.configure({ adapter: new EnzymeAdapter() });
const setup = (props={},state=null) => {
  return shallow(
    <Provider store ={store}>
            <ProjectResult {...props} />
    </Provider>

  );
};
const findByTestAttr = (wrapper, val) => {
    return wrapper.find(`[data-test="${val}"]`);
};
describe(" test of ProjectResult page ",()=>{
    test("render without error",()=>{
        let wrapper = setup();
        expect(wrapper).toBeTruthy();
        expect(wrapper.length).toEqual(1);
    })

    test('show projects : ', () => {
        const wrapper = setup();
        let field = findByTestAttr(wrapper, 'showProjects')
        expect(field.length).toBe(0);
        expect(field.text).toBeTruthy();
    })

    test('show counter: ', () => {
        let wrapper = setup();
        wrapper.setProps({
            location: 'string',
            page:'number'
          })
        expect(wrapper.length).not.toEqual(0);
    })

    it('check props not render without class',()=>{
        let wrapper = setup();

         let props = {
             label: 'button', onClick: () => { }
         }

          wrapper.setProps(props)

          expect(wrapper.props()).not.toEqual(props);
          expect(wrapper.props().label).toBeUndefined()

    })

    it('simulate button',()=>{
        let wrapper = setup();
        const mockCallBack = sinon.spy();
        let props = {
            label: 'button', onClick: mockCallBack
        }
        wrapper.setProps(props)
        wrapper.update();
        expect(mockCallBack).toHaveProperty('callCount', 0);
    })

    test('snapshot', () => {
      let wrapper =(<BrowserRouter>
      <Provider store={store}>
        <ProjectResult/>
      </Provider></BrowserRouter>)
      const snapshot = renderer.create(wrapper).toJSON();
      expect(snapshot).toMatchSnapshot();
  })

});

describe('integration test component', () => {
let mockProject_1={
    amount: 5000009,
    category: "سرور",
    categoryId: 72,
    creator: "سوپریوزر سوپریوزر",
    creatorUrl: "/users/superuser/",
    description: "نیاز به برقراری ارتباط با سرور جزو بزرگ ترین خواسته های ما است.",
    endDate: "2021-02-09T22:17:47.204072+03:30",
    name: "تکاوری",
    skills: [{id: 75, name: "containers"}, {id: 74, name: "Nginx"}, {id: 73, name: "لینوکس"}],
    status: {code: "ENDED", label: "پایان یافته"},
    url: "/projects/%D8%AA%DA%A9%D8%A7%D9%88%D8%B1%DB%8C/"
}
let store_mock
    beforeEach(() => {
      store_mock = mockStore({
        search: {
            projects: [mockProject_1,mockProject_1 ],
            text: ""
        },
        userProfile: {
          loading: true,
          provinces: [],
          cities: [],
          universities: [],
          profile: {},
          description: '',
          data: {},
          parentSkills: [],
          skills: [],
          resume: null,
          avatar: null,
          briefprofile: {}
        }
      });
    });
  
    test("mock store check projects", () => {
      const dom = render(
        <BrowserRouter>
          <Provider store={store_mock}>
            <ProjectResult />
          </Provider>
        </BrowserRouter>
      );
      expect(screen.getAllByTestId("project-card").length).toEqual(2)
    })
  
  
    test("change search input", async () => {
      const spySetSearchText = jest.spyOn(actions, 'setSearchText');
      store_mock.dispatch = jest.fn()
      const dom = render(
        <BrowserRouter>
          <Provider store={store_mock}>
            <ProjectResult />
          </Provider>
        </BrowserRouter>
      );
  
      fireEvent.change(screen.getByTestId('search-input').querySelector('input'), { target: { value: 'test' } });
      expect(screen.getByTestId('search-input').querySelector('input').value).toBe('test');
  
      //check dispatch one setSearchText after input change
      await wait(() => {
        expect(spySetSearchText).toHaveBeenCalledTimes(1);
      });
    })
  
    test("change pagination ", async () => {

      const spyChangePagination = ProjectResult.prototype.handleChange = jest.fn();
      const spySearchPage = jest.spyOn(actions, 'searchPage');
      store_mock.dispatch = jest.fn()
      const dom = render(
        <BrowserRouter>
          <Provider store={store_mock}>
            <ProjectResult />
          </Provider>
        </BrowserRouter>
      );
  
      await spyChangePagination(Event, 2)
  
      //check dispatch one setSearchText after input change
      await wait(() => {
        //test call handleChange
        expect(spyChangePagination).toHaveBeenCalledTimes(1);
        expect(spyChangePagination).toHaveBeenCalledWith(Event, 2);
        //test call searchUsersPage
        expect(spySearchPage).toHaveBeenCalledTimes(1);
        expect(spySearchPage).toHaveBeenCalledWith(1)
      });
    })
  
    
  })