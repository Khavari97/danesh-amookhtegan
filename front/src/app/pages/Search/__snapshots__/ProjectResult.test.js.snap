// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` test of ProjectResult page  snapshot 1`] = `
Array [
  <div
    className="MuiGrid-root MuiGrid-item"
    style={
      Object {
        "display": "flex",
        "margin": "5px 0 40px 0",
        "padding": 5,
        "textAlign": "center",
      }
    }
  >
    <div
      className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth"
      data-test="placeholder"
      data-testid="search-input"
      onClick={[Function]}
    >
      <input
        className="MuiInputBase-input MuiInput-input"
        defaultValue=""
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        placeholder="مهارت یا اسم پروژه یا نام کاربری فرد مورد نظر خود را وارد کنید"
        type="text"
      />
    </div>
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
      data-test="btnSearch"
      disabled={false}
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "marginRight": 5,
        }
      }
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        جستجو
        <img
          alt=""
          src="search.svg"
          style={
            Object {
              "marginRight": 5,
            }
          }
          width="100%"
        />
      </span>
      <span
        className="MuiTouchRipple-root"
      />
    </button>
  </div>,
  <div
    className="MuiGrid-root"
  >
    <div
      className="MuiGrid-root MuiGrid-container"
    >
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-sm-12"
        data-test="sorting"
        style={
          Object {
            "padding": "0 5px 15px 0",
          }
        }
      >
         دسته بندی بر اساس : 
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-4"
      >
        <div
          aria-expanded={false}
          aria-owns={null}
          className="MuiAutocomplete-root MuiAutocomplete-hasClearIcon MuiAutocomplete-hasPopupIcon"
          data-test="Autocomplete"
          data-testid=" status "
          onClick={[Function]}
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          role="combobox"
          style={
            Object {
              "marginBottom": "20px",
              "padding": "5px 5px",
            }
          }
        >
          <div
            className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
            data-test="field"
          >
            <label
              className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-marginDense MuiInputLabel-outlined"
              data-shrink={false}
              htmlFor="combo-box-demo"
              id="combo-box-demo-label"
            >
               وضعیت 
            </label>
            <div
              className="MuiInputBase-root MuiOutlinedInput-root MuiAutocomplete-inputRoot MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd MuiOutlinedInput-adornedEnd MuiInputBase-marginDense MuiOutlinedInput-marginDense"
              onClick={[Function]}
            >
              <input
                aria-activedescendant={null}
                aria-autocomplete="list"
                aria-controls={null}
                aria-invalid={false}
                autoCapitalize="none"
                autoComplete="off"
                autoFocus={false}
                className="MuiInputBase-input MuiOutlinedInput-input MuiAutocomplete-input MuiAutocomplete-inputFocused MuiInputBase-inputAdornedEnd MuiOutlinedInput-inputAdornedEnd MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
                disabled={false}
                id="combo-box-demo"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                onMouseDown={[Function]}
                required={false}
                spellCheck="false"
                type="text"
                value=""
              />
              <div
                className="MuiAutocomplete-endAdornment"
              >
                <button
                  aria-label="Clear"
                  className="MuiButtonBase-root MuiIconButton-root MuiAutocomplete-clearIndicator"
                  disabled={false}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onDragLeave={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  tabIndex={-1}
                  title="Clear"
                  type="button"
                >
                  <span
                    className="MuiIconButton-label"
                  >
                    <svg
                      aria-hidden={true}
                      className="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                      />
                    </svg>
                  </span>
                  <span
                    className="MuiTouchRipple-root"
                  />
                </button>
                <button
                  aria-label="Open"
                  className="MuiButtonBase-root MuiIconButton-root MuiAutocomplete-popupIndicator"
                  disabled={false}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onDragLeave={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  tabIndex={-1}
                  title="Open"
                  type="button"
                >
                  <span
                    className="MuiIconButton-label"
                  >
                    <svg
                      aria-hidden={true}
                      className="MuiSvgIcon-root"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M7 10l5 5 5-5z"
                      />
                    </svg>
                  </span>
                  <span
                    className="MuiTouchRipple-root"
                  />
                </button>
              </div>
              <fieldset
                aria-hidden={true}
                className="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
              >
                <legend
                  className="PrivateNotchedOutline-legendLabelled-3"
                >
                  <span>
                     وضعیت 
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-4"
      >
        <div
          aria-expanded={false}
          aria-owns={null}
          className="MuiAutocomplete-root MuiAutocomplete-hasClearIcon MuiAutocomplete-hasPopupIcon"
          data-test="Autocomplete"
          data-testid=" parentSkill "
          onClick={[Function]}
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          role="combobox"
          style={
            Object {
              "marginBottom": "20px",
              "padding": "5px 5px",
            }
          }
        >
          <div
            className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
            data-test="field"
          >
            <label
              className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-marginDense MuiInputLabel-outlined"
              data-shrink={false}
              htmlFor="combo-box-demo"
              id="combo-box-demo-label"
            >
               دسته بندی 
            </label>
            <div
              className="MuiInputBase-root MuiOutlinedInput-root MuiAutocomplete-inputRoot MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd MuiOutlinedInput-adornedEnd MuiInputBase-marginDense MuiOutlinedInput-marginDense"
              onClick={[Function]}
            >
              <input
                aria-activedescendant={null}
                aria-autocomplete="list"
                aria-controls={null}
                aria-invalid={false}
                autoCapitalize="none"
                autoComplete="off"
                autoFocus={false}
                className="MuiInputBase-input MuiOutlinedInput-input MuiAutocomplete-input MuiAutocomplete-inputFocused MuiInputBase-inputAdornedEnd MuiOutlinedInput-inputAdornedEnd MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
                disabled={false}
                id="combo-box-demo"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                onMouseDown={[Function]}
                required={false}
                spellCheck="false"
                type="text"
                value=""
              />
              <div
                className="MuiAutocomplete-endAdornment"
              >
                <button
                  aria-label="Clear"
                  className="MuiButtonBase-root MuiIconButton-root MuiAutocomplete-clearIndicator"
                  disabled={false}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onDragLeave={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  tabIndex={-1}
                  title="Clear"
                  type="button"
                >
                  <span
                    className="MuiIconButton-label"
                  >
                    <svg
                      aria-hidden={true}
                      className="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                      />
                    </svg>
                  </span>
                  <span
                    className="MuiTouchRipple-root"
                  />
                </button>
                <button
                  aria-label="Open"
                  className="MuiButtonBase-root MuiIconButton-root MuiAutocomplete-popupIndicator"
                  disabled={false}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onDragLeave={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  tabIndex={-1}
                  title="Open"
                  type="button"
                >
                  <span
                    className="MuiIconButton-label"
                  >
                    <svg
                      aria-hidden={true}
                      className="MuiSvgIcon-root"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M7 10l5 5 5-5z"
                      />
                    </svg>
                  </span>
                  <span
                    className="MuiTouchRipple-root"
                  />
                </button>
              </div>
              <fieldset
                aria-hidden={true}
                className="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
              >
                <legend
                  className="PrivateNotchedOutline-legendLabelled-3"
                >
                  <span>
                     دسته بندی 
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-4"
      >
        <div
          aria-expanded={false}
          aria-owns={null}
          className="MuiAutocomplete-root MuiAutocomplete-hasClearIcon MuiAutocomplete-hasPopupIcon"
          data-test="Autocomplete"
          data-testid=" childSkill "
          onClick={[Function]}
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          role="combobox"
          style={
            Object {
              "marginBottom": "20px",
              "padding": "5px 5px",
            }
          }
        >
          <div
            className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
            data-test="field"
          >
            <label
              className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-marginDense MuiInputLabel-outlined"
              data-shrink={false}
              htmlFor="combo-box-demo"
              id="combo-box-demo-label"
            >
               مهارت 
            </label>
            <div
              className="MuiInputBase-root MuiOutlinedInput-root MuiAutocomplete-inputRoot MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd MuiOutlinedInput-adornedEnd MuiInputBase-marginDense MuiOutlinedInput-marginDense"
              onClick={[Function]}
            >
              <input
                aria-activedescendant={null}
                aria-autocomplete="list"
                aria-controls={null}
                aria-invalid={false}
                autoCapitalize="none"
                autoComplete="off"
                autoFocus={false}
                className="MuiInputBase-input MuiOutlinedInput-input MuiAutocomplete-input MuiAutocomplete-inputFocused MuiInputBase-inputAdornedEnd MuiOutlinedInput-inputAdornedEnd MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
                disabled={false}
                id="combo-box-demo"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                onMouseDown={[Function]}
                required={false}
                spellCheck="false"
                type="text"
                value=""
              />
              <div
                className="MuiAutocomplete-endAdornment"
              >
                <button
                  aria-label="Clear"
                  className="MuiButtonBase-root MuiIconButton-root MuiAutocomplete-clearIndicator"
                  disabled={false}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onDragLeave={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  tabIndex={-1}
                  title="Clear"
                  type="button"
                >
                  <span
                    className="MuiIconButton-label"
                  >
                    <svg
                      aria-hidden={true}
                      className="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                      />
                    </svg>
                  </span>
                  <span
                    className="MuiTouchRipple-root"
                  />
                </button>
                <button
                  aria-label="Open"
                  className="MuiButtonBase-root MuiIconButton-root MuiAutocomplete-popupIndicator"
                  disabled={false}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onDragLeave={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  tabIndex={-1}
                  title="Open"
                  type="button"
                >
                  <span
                    className="MuiIconButton-label"
                  >
                    <svg
                      aria-hidden={true}
                      className="MuiSvgIcon-root"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M7 10l5 5 5-5z"
                      />
                    </svg>
                  </span>
                  <span
                    className="MuiTouchRipple-root"
                  />
                </button>
              </div>
              <fieldset
                aria-hidden={true}
                className="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
              >
                <legend
                  className="PrivateNotchedOutline-legendLabelled-3"
                >
                  <span>
                     مهارت 
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  <div
    className="MuiGrid-root animate__animated animate__fadeIn MuiGrid-container MuiGrid-spacing-xs-2"
  >
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
      style={
        Object {
          "marginTop": "50px",
        }
      }
    >
      <svg
        height="40"
        viewBox="0 0 65 65.235"
        width="40"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g
          data-name="Group 5"
          id="Group_5"
          transform="translate(-1574 -446)"
        >
          <path
            d="M7.531,0A7.4,7.4,0,0,0,0,7.531V52.718a7.4,7.4,0,0,0,7.531,7.531H37.42a17.4,17.4,0,0,1-7.06-5.021H7.531a2.368,2.368,0,0,1-2.51-2.51V10.041H55.228V30.36a15.856,15.856,0,0,1,5.021,7.06V7.531A7.4,7.4,0,0,0,52.718,0Zm5.021,20.083V25.1h5.021V20.083Zm10.041,0V25.1H47.7V20.083ZM12.552,30.124v5.021h5.021V30.124Zm10.041,0v5.021h6.041a13.714,13.714,0,0,1,4.236-5.021Zm21.338,0a13.807,13.807,0,1,0,0,27.614A13.611,13.611,0,0,0,52.4,54.679l.471.471a2.356,2.356,0,0,0,.314,3.059l6.276,6.276a2.452,2.452,0,0,0,3.53,0L64.25,63.23a2.361,2.361,0,0,0,0-3.452L57.974,53.5a2.4,2.4,0,0,0-2.9-.471l-.471-.471a13.635,13.635,0,0,0,3.138-8.629A13.93,13.93,0,0,0,43.931,30.124Zm0,3.766A10.041,10.041,0,1,1,33.89,43.931,10.074,10.074,0,0,1,43.931,33.89Zm-31.38,6.276v5.021h5.021V40.166Zm10.041,0v5.021h3.766V43.931a19.014,19.014,0,0,1,.471-3.766Z"
            fill="#046b00"
            id="icons8_search_property"
            transform="translate(1574 446)"
          />
          <g
            data-name="Group 4"
            id="Group_4"
            transform="translate(162 113)"
          >
            <line
              data-name="Line 3"
              fill="none"
              id="Line_3"
              stroke="#046b00"
              strokeLinecap="round"
              strokeWidth="2"
              transform="translate(1450.5 372.5)"
              x2="10"
              y2="9"
            />
            <line
              data-name="Line 4"
              fill="none"
              id="Line_4"
              stroke="#046b00"
              strokeLinecap="round"
              strokeWidth="2"
              transform="translate(1450.5 372.5)"
              x1="10"
              y2="9"
            />
          </g>
        </g>
      </svg>
      <p
        className="MuiTypography-root MuiTypography-body1 MuiTypography-colorPrimary"
        style={
          Object {
            "marginRight": "10px",
          }
        }
      >
        پروژه ای با این مشخصات یافت نشد!
      </p>
    </div>
  </div>,
  <div
    className="MuiGrid-root"
    style={
      Object {
        "display": "table",
        "margin": "0 auto",
        "padding": "40px 0",
      }
    }
  >
    <nav
      aria-label="pagination navigation"
      className="MuiPagination-root"
    >
      <ul
        className="MuiPagination-ul"
      >
        <li>
          <button
            aria-label="Go to previous page"
            className="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-page MuiPaginationItem-outlined MuiPaginationItem-rounded MuiPaginationItem-outlinedPrimary Mui-disabled MuiPaginationItem-sizeLarge Mui-disabled"
            disabled={true}
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={-1}
            type="button"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiPaginationItem-icon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
              />
            </svg>
          </button>
        </li>
        <li>
          <button
            aria-label="Go to next page"
            className="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-page MuiPaginationItem-outlined MuiPaginationItem-rounded MuiPaginationItem-outlinedPrimary Mui-disabled MuiPaginationItem-sizeLarge Mui-disabled"
            disabled={true}
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={-1}
            type="button"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiPaginationItem-icon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
              />
            </svg>
          </button>
        </li>
      </ul>
    </nav>
  </div>,
]
`;
